### PR TITLE
feat: companion

### PR DESCRIPTION
**Pull request by rebelchris**
Introducing the changes needed for the companion app

- optOutCompanion setting
- getPostByCanonicalUrl query

dd-458

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -179,6 +179,9 @@ export default async function app(
             extendKey: trackingExtendKey,
           },
           banner: true,
+          postCanonical: {
```
Comment (idoshamun) [inline src/index.ts:182]: Nice!

File: src/schema/posts.ts [hunk 2]
```diff
@@ -304,6 +304,16 @@ export const typeDefs = /* GraphQL */ `
       id: ID
     ): Post!
 
+    """
+    Get post by canonical URL
+    """
+    postCanonical(
```
Comment (idoshamun) [inline src/schema/posts.ts:310]: I would suggest this name which is a bit more explicit in my opinion.
I also think we should compare both canonical and the original URL to the given parameter.
What do you think?

```suggestion
    postByUrl(
```

File: src/schema/posts.ts [hunk 3]
```diff
@@ -457,6 +467,24 @@ export const resolvers: IResolvers<any, Context> = {
       }
       throw new NotFoundError('Post not found');
     },
+    postCanonical: async (
+      source,
+      { canonicalUrl }: { id: string; canonicalUrl: string },
+      ctx: Context,
+      info,
+    ) => {
+      const res = await graphorm.query(ctx, info, (builder) => ({
```
Comment (idoshamun) [inline src/schema/posts.ts:476]: Please add a limit 1 to this query should be faster (a bit).

**Review by idoshamun — APPROVED** [review]
Looking good! A few suggestions for improvement and we can merge 😬

File: src/schema/posts.ts (hunk 2 above)
Comment (rebelchris) [reply src/schema/posts.ts:310]: Good point, my idea was that the canonical would be leading anyway, but happy to include a URL or Canonical as the query.

**Comment by rebelchris** [comment]
@idoshamun Switched to postByUrl, added both test cases for URL and Canonical URL.

**Review by idoshamun — APPROVED** [review]
Awesome!

**Pull request merged**